### PR TITLE
Verena WDFT Berlin June 2020

### PR DESCRIPTION
**Pull request by verenamay**

### CODE REVIEW

File: src/functions-and-arrays.js [hunk 1]
```diff
@@ -1,19 +1,77 @@
...
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(numbers) {
+  let total=0;
+  for(var i in numbers) {
+    total +=numbers[i];
+    
+  } return total
+}
```
Comment (Jeff-Moraes) [inline src/functions-and-arrays.js:49]: great solution ;)

File: src/functions-and-arrays.js [hunk 2]
```diff
@@ -1,19 +1,77 @@
...
+    if(words[j].length == tlength){
+       wholeArr.push(words[j]);
+    }
+  }
+ if(wholeArr.length == 1){
+   return wholeArr[0]
+ }else{
+    return wholeArr
+}
+}
```
Comment (Jeff-Moraes) [inline src/functions-and-arrays.js:36]: the logic that you used is on the right way, but it is a bit simpler than what you tried.
here is one solution for this interaction
```
let longestWord = "";
  for (let i = 0; i < words.length; i++) {
    if (words[i].length > longestWord.length) {
      longestWord = words[i];
    }
  }
  return longestWord;
```

File: src/functions-and-arrays.js [hunk 3]
```diff
@@ -1,19 +1,77 @@
...
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(numbersAvg) {
+  let total=0;
+  for(var i in numbersAvg) {
+    total +=numbersAvg[i];
+} let average=0;
+average = total/numbersAvg.length; 
+return average;
+}
```
Comment (Jeff-Moraes) [inline src/functions-and-arrays.js:63]: to solve this error
`should return null if receives an empty array when called`
you can just check if the numbersAvg that you are receiving in the function has some value inside of it
```
if (numbersAvg.length === 0) return null;
```

File: src/functions-and-arrays.js [hunk 4]
```diff
@@ -1,19 +1,77 @@
...
+
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
+function averageWordLength(wordsArr) {
+  avg = wordsArr.join('').length/wordsArr.length;
+  return avg
+}
```
Comment (Jeff-Moraes) [inline src/functions-and-arrays.js:71]: same here if the wordsArr is empty
```
if (wordsArr.length === 0) return null;
```

Comment (verenamay) [reply src/functions-and-arrays.js:71]: Thanks for your feedback, I will try the other iterations on my own as well :)

**Pull request closed without merging**